pkg/controllers/state/statenode.go

```diff
@@ -312,6 +318,18 @@ func (in *StateNode) Initialized() bool {
 	return true
 }
 
+func (in *StateNode) EventualDisruptionCandidate() bool {
+	if !in.Managed() {
+		return false
+	}
+	return lo.ContainsBy(in.NodeClaim.GetConditions(), func(cond status.Condition) bool {
+		if cond.Type != v1beta1.ConditionTypeDrifted && cond.Type != v1beta1.ConditionTypeExpired {
```

Comment:
Does empty not come into scope here?

Reply:
No, we only want to weigh away from drifted or expired nodes. If we can reschedule to an empty node, that is the indication that we **shouldn't** consolidate an empty node.